contracts/StoryProtocolGateway.sol

```diff
@@ -1,503 +1,424 @@
+    function setNftContractBeacon(address newNftContractBeacon) external restricted {
+        if (newNftContractBeacon == address(0)) revert Errors.SPG__ZeroAddressParam();
+        SPGStorage storage $ = _getSPGStorage();
+        $.nftContractBeacon = newNftContractBeacon;
+    }
 
-        if (policyId != 0) {
-            LICENSING_MODULE.addPolicyToIp(ipId, policyId);
-        }
-        _setCustomIpMetadata(ipId, ipMetadata.customMetadata);
+    /// @dev Upgrades the NFT contract beacon. Restricted to only the protocol access manager.
+    /// @param newNftContract The address of the new NFT contract implemenetation.
+    function upgradeCollections(address newNftContract) public restricted {
+        // UpgradeableBeacon checks for newImplementation.bytecode.length > 0, so no need to check for zero address.
+        UpgradeableBeacon(_getSPGStorage().nftContractBeacon).upgradeTo(newNftContract);
     }
 
-    /// @notice Mints a Story Protocol NFT and registers it into the protocol as an IP asset with user signature.
-    /// @dev This function allows the user to set the permission for the SPG with a
-    /// signature to allow SPG call other modules like licensing module on behalf of the user.
-    /// @param tokenContract The address of the NFT bound to the root-level IP.
-    /// @param tokenMetadata Additional token metadata in bytes to include for minting.
-    /// @param ipMetadata Metadata related to IP attribution.
-    /// @param signature The signature to set the permission for the IP.
-    function mintAndRegisterIpWithSig(
-        uint256 policyId,
-        address tokenContract,
-        bytes calldata tokenMetadata,
-        Metadata.IPMetadata calldata ipMetadata,
-        SPG.Signature calldata signature
-    ) external returns (uint256 tokenId, address ipId) {
-        tokenId = _mint(tokenContract, tokenMetadata, msg.sender);
-        ipId = _registerIp(tokenContract, tokenId, ipMetadata.name, ipMetadata.hash, ipMetadata.url);
-
-        _setPermissionWithSig(ipId, signature.signer, signature.deadline, signature.signature);
-
-        if (policyId != 0) {
-            LICENSING_MODULE.addPolicyToIp(ipId, policyId);
-        }
-        _setCustomIpMetadata(ipId, ipMetadata.customMetadata);
+    /// @notice Creates a new NFT collection to be used by SPG.
+    /// @param name The name of the collection.
+    /// @param symbol The symbol of the collection.
+    /// @param maxSupply The maximum supply of the collection.
+    /// @param mintCost The cost to mint an NFT from the collection.
+    /// @param mintToken The token to be used for mint payment.
+    /// @param owner The owner of the collection.
+    /// @return nftContract The address of the newly created NFT collection.
+    function createCollection(
+        string calldata name,
+        string calldata symbol,
+        uint32 maxSupply,
+        uint256 mintCost,
+        address mintToken,
```

Comment:
Consider rename to mintPaymentToken or mintFeeToken for better readability.

Reply:
Resolved in 58ba77b